frontend/src/modules/history/components/Translation.js

```diff
@@ -0,0 +1,99 @@
+/* @flow */
+
+import * as React from 'react';
+
+import './Translation.css';
+
+import type { DBTranslation } from '../reducer';
+
+
+type Props = {|
+    translation: DBTranslation,
+|};
+
+/**
+ * Render a translation in the History tab.
+ *
+ * Shows the translation's status, date, author and reviewer, as well as
+ * the content of the translation.
+ *
+ * The status can be interact with if the user has sufficient permissions to
+ * change said status.
+ */
+export default class Translation extends React.Component<Props> {
+    getStatus() {
+        const { translation } = this.props;
+
+        if (translation.approved) {
+            return 'approved';
+        }
+        if (translation.fuzzy) {
+            return 'fuzzy';
+        }
+        if (translation.rejected) {
+            return 'rejected';
+        }
+        return 'unreviewed';
+    }
+
+    getApprovalTitle() {
+        const { translation } = this.props;
+
+        if (translation.approved && translation.approved_user) {
+            return 'Approved by ' + translation.approved_user;
```

Comment:
It would probably be better to add a level of indirection between what the API returns and the attributes we use in our code. Here for example we should use `translation.approvedUser` and in the reducer get that from `translation.approved_user`.

Reply:
I also need to see if it's better to use `'a ' + b` or `` `a ${b}` ``.

Reply:
They won't be better or worse unless you are microbenchmarking ;)

Reply:
Nice! Is there one that is more commonly used by the JS community?

Reply:
Ever since they are available, I see template strings used widely. Their usefulness shines as more parameters are used as part of the string.

In your code above you'll want to avoid string concatenation if you ever plan to support i18n anyway.